### PR TITLE
Run genjsonschema in CI

### PR DESCRIPTION
**Pull request by dmathieu**
https://github.com/open-telemetry/opentelemetry-go-contrib/pull/6347 is updating the generated JSON schema, as it has changed with upgrades of `go-jsonschema`.
We update the binary semi automatically with dependabot. However, this means there could be unseen regressions or changes like this that we miss.

This runs the schema generation on CI, before checking for a clean repository, so we can keep the generated code up to date with the binary version.

This should fail until #6347 and #6349 are merged.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (pellared) [inline .github/workflows/ci.yml:48]: Why not adding it to `generate` target in Makefile?

Comment (dmathieu) [reply .github/workflows/ci.yml:48]: I hesitated, and I can do it.
I went with not doing it because the command takes a bit of time to run, and needs to be executed much less often than generate.

Comment (pellared) [reply .github/workflows/ci.yml:48]: Reasonable to me if it takes more than a few seconds.
Otherwise I would add it to `generate` target.

**Review by pellared — APPROVED** [review]

File: .github/workflows/ci.yml
Comment (dmathieu) [reply .github/workflows/ci.yml:48]: The command runs the linter. If we remove it, running it on generate seems fine

Comment (pellared) [reply .github/workflows/ci.yml:48]: Makes sense. I think not running linter will also be clenaer.

**Review by evan-bradley — APPROVED** [review]
Thanks! Wasn't sure where to add this when updating the types, but this is the obvious next step to make sure they don't go out-of-date.

**Review by pellared — APPROVED** [review]

**Comment by pellared** [comment]
CC @codeboten

**Comment by dmathieu** [comment]
@open-telemetry/go-approvers this needs a second review.

**Review by codeboten — APPROVED** [review]

File: Makefile
Comment (MrAlias) [reply Makefile:345]: I don't follow. We don't run `generate` on `lint`, but we do run `lint` at the end of `genjsonschema`. Why is this changing now that we are running `genjsonschema` in our CI?

Comment (dmathieu) [reply Makefile:345]: Ah, I see the misunderstanding. I messed up my updates from the discussion above.

With my latest commit, `jsonschemagen` now runs alongside `generate`. We therefore don't want to run `lint`, because it's already being run in `precommit` and `ci`.

**Review by MrAlias — APPROVED** [review]

**Comment by dmathieu** [comment]
This has two reviews from approvers from the same employer.
However, it's also reviewed by the code owner of the `config` package. I am therefore going to merge it.

**Pull request merged**
